basiclti/basiclti-blis/src/java/org/sakaiproject/blti/ServiceServlet.java

```diff
@@ -641,6 +642,14 @@ protected void processRoster(HttpServletRequest request, HttpServletResponse res
 			Set<Member> members = site.getMembers();
 			Map<String, String> roleMap = SakaiBLTIUtil.convertRoleMapPropToMap(roleMapProp);
 			for (Member member : members ) {
+				User user;
+				try {
+					user = UserDirectoryService.getUser(member.getUserId());
```

Comment:
Can you make use of the much faster JLDAP getUsers method that will result in one LDAP call instead of N?